activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -103,7 +103,7 @@ def visit_Arel_Nodes_ValuesList(o, collector)
             row.each_with_index do |value, k|
               collector << ", " unless k == 0
               case value
-              when Nodes::SqlLiteral, Nodes::BindParam, ActiveModel::Attribute
+              when Nodes::Node, ActiveModel::Attribute
```

Comment:
SqlLiteral isn't a Node, because of reasons.

This should probably technically use `Arel.arel_node?`, but just keeping that class in this list would also be maybe-fine.

Reply:
Oh right, it's a `String` and not a subclass of `Arel::Nodes::Node`. Surprisingly it still worked because `quote` is explicitly handling `Nodes::SqlLiteral`. 

Nonetheless, I added a test for the behavior and added `Nodes::SqlLiteral` back to the list.